client/client_test.go

```diff
@@ -228,6 +232,21 @@ func (s *clientTestSuite) TestConn_SetCharset() {
 	require.NoError(s.T(), err)
 }
 
+func (s *clientTestSuite) TestConn_SetCollationAfterConnect() {
+	err := s.c.SetCollation("latin1_swedish_ci")
+	require.Error(s.T(), err)
```

Comment:
we can also check the error message is readable by `require.ErrorContains`. Not a big problem